presto-spark/src/main/java/com/facebook/presto/spark/TaskProcessors.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spark;
+
+import org.apache.spark.TaskContext;
+import org.apache.spark.api.java.function.FlatMapFunction2;
+import org.apache.spark.api.java.function.PairFlatMapFunction;
+import org.apache.spark.util.CollectionAccumulator;
+import scala.Tuple2;
+
+import java.util.HashMap;
+import java.util.Iterator;
+
+import static java.util.Collections.emptyMap;
+import static java.util.Collections.singletonMap;
+import static java.util.Collections.unmodifiableMap;
+
+public class TaskProcessors
+{
+    private TaskProcessors() {}
+
+    public static PairFlatMapFunction<byte[], Integer, byte[]> createTaskProcessor(
```

Comment:
I am wondering if we should just inline these methods. The type looks so complicated and it looks like only used once.

Reply:
I was trying to do this in the beginning. Unfortunately there are two main problems with that approach:

1. Lambda capturing:
When an outer class field is referenced from within the lambda, the outer class reference is captured. For example: 
```
class Outer {
 Serializable object

 ... function() {
   ...
   () - > { print(object) }
   ...
 }
}
``` 
In this example the `Outer` reference is captured (instead of direct reference to `object`). Since `Outer` is not serialize the execution will fail. 
We can try to avoid referencing any outer fields, but it is very fragile.

2. Lambda serialization. Lambda declared in a child class loader (which Presto claasloader is) cannot be deserialized. The deserialization code is run in the main (Spark) class loader. And classes defined in the Presto classloader are not visible from there. Thus during the deserialization Spark classloader doesn't know anything about the class, and cannot create an instance of it.

Reply:
@arhimondr : 

> Lambda serialization. Lambda declared in a child class loader (which Presto claasloader is) cannot be deserialized. The deserialization code is run in the main (Spark) class loader. And classes defined in the Presto classloader are not visible from there. Thus during the deserialization Spark classloader doesn't know anything about the class, and cannot create an instance of it.

Good point! That's why we need static utility class :)